app/controllers/reviews_controller.rb

```diff
@@ -11,15 +12,48 @@ def create
     redirect_to item_path(@item)
   end
 
+  def destroy
+    @review = Review.find(params[:item_id])
+    @item = @review.item
+    if @review.user == current_user
+      @review.destroy
+    end
+    redirect_to item_path(@item)
+  end
+
+  def edit
+    @review = Review.find(params[:review_id])
+    @item = Item.find(@review.item.id)
+    if current_user == @review.user
+      @rating_options = Review::RATING_OPTIONS
+    else
+      flash[:error] = "You are not authorized to view this page!"
+      redirect_to root_path
+    end
+  end
```

Comment:
You can clean up some of this logic using filters:

http://guides.rubyonrails.org/action_controller_overview.html#filters